geode-core/src/integrationTest/java/org/apache/geode/internal/ClassPathLoaderIntegrationTest.java

```diff
@@ -554,22 +554,11 @@ protected File getTempFile() {
     }
   }
 
-  /**
-   * Custom class loader which uses BCEL to always dynamically generate a class for any class name
-   * it tries to load.
-   */
-  private class GeneratingClassLoader2 extends GeneratingClassLoader {
-    @Override
-    protected File getTempFile() {
-      return tempFile2;
-    }
-  }
-
   private File createJarWithClass(String fileName, String className) throws IOException {
     String stringBuilder = "package integration.parent;" + "public class " + className + " {}";
     File jarFile = new File(temporaryFolder.getRoot(), fileName);
     if (jarFile.exists()) {
-      jarFile.delete();
+      jarFile = new File(temporaryFolder.newFolder(), fileName);
```

Comment:
I think the root of the problem is that the tests use the same dir for staging as for deploying. I would prefer that the tests _always_ use separate dirs for staging and deploying.